apis/infrastructure/v1beta1/byohost_webhook.go

```diff
@@ -4,48 +4,44 @@
 package v1beta1
 
 import (
-	"errors"
-
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	ctrl "sigs.k8s.io/controller-runtime"
-	logf "sigs.k8s.io/controller-runtime/pkg/log"
-	"sigs.k8s.io/controller-runtime/pkg/webhook"
-)
-
-// log is for logging in this package
-var byohostlog = logf.Log.WithName("byohost-resource")
+	"context"
+	"net/http"
 
-// SetupWebhookWithManager sets up the webhook for the byohost resource
-func (byoHost *ByoHost) SetupWebhookWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewWebhookManagedBy(mgr).
-		For(byoHost).
-		Complete()
-}
+	v1 "k8s.io/api/admission/v1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
 
 //+kubebuilder:webhook:path=/validate-infrastructure-cluster-x-k8s-io-v1beta1-byohost,mutating=false,failurePolicy=fail,sideEffects=None,groups=infrastructure.cluster.x-k8s.io,resources=byohosts,verbs=create;update;delete,versions=v1beta1,name=vbyohost.kb.io,admissionReviewVersions={v1,v1beta1}
 
-var _ webhook.Validator = &ByoHost{}
-
-// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
-func (byoHost *ByoHost) ValidateCreate() error {
-	return nil
+// +k8s:deepcopy-gen=false
+// ByohHostValidator validates ByoHosts
+type ByohHostValidator struct {
```

Comment:
Nit only: Just to keep it in sync
```suggestion
type ByoHostValidator struct {
```